pyresample/utils.py

```diff
@@ -124,53 +146,60 @@ def _parse_yaml_area_file(area_file_name, *regions):
     the files, using the first file as the "base", replacing things after
     that.
     """
-    from pyresample.geometry import DynamicAreaDefinition
     area_dict = _read_yaml_area_file_content(area_file_name)
     area_list = regions or area_dict.keys()
-
     res = []
-
     for area_name in area_list:
-        try:
-            params = area_dict[area_name]
-        except KeyError:
-            raise AreaNotFound('Area "{0}" not found in file "{1}"'.format(
-                area_name, area_file_name))
-        description = params['description']
-        projection = params['projection']
-        optimize_projection = params.get('optimize_projection', False)
-        try:
-            xsize = params['shape']['width']
-            ysize = params['shape']['height']
-        except KeyError:
-            xsize, ysize = None, None
-        try:
-            area_extent = (params['area_extent']['lower_left_xy'] +
-                           params['area_extent']['upper_right_xy'])
-        except KeyError:
-            area_extent = None
-        try:
-            projection['units'] = params['area_extent']['units']
-        except KeyError:
-            pass
-        try:
-            rotation = params['rotation']
-        except KeyError:
-            rotation = 0
-        area = DynamicAreaDefinition(area_name, description,
-                                     projection, xsize, ysize,
-                                     area_extent,
-                                     optimize_projection,
-                                     rotation)
-        try:
-            area = area.freeze()
-        except (TypeError, AttributeError):
-            pass
-
-        res.append(area)
+        params = area_dict.get(area_name)
+        if params is None:
+            raise AreaNotFound('Area "{0}" not found in file "{1}"'.format(area_name, area_file_name))
+        params.setdefault('area_id', area_name)
+        # Optional arguments.
+        params['shape'] = _get_list(params, 'shape', ['height', 'width'])
+        params['upper_left_extent'] = _get_list(params, 'upper_left_extent', ['x', 'y'])
+        params['center'] = _get_list(params, 'center', ['x', 'y'])
+        params['area_extent'] = _get_list(params, 'area_extent', ['lower_left_xy', 'upper_right_xy'])
+        params['resolution'] = _get_list(params, 'resolution', ['dx', 'dy'])
+        params['radius'] = _get_list(params, 'radius', ['dx', 'dy'])
+        params['rotation'] = _get_list(params, 'rotation', [])
+        res.append(from_params(**params))
     return res
 
 
+def _get_list(params, var, arg_list):
```

Comment:
`params` isn't used in this function appart from getting `var` from it. I would rather have it called like `_get_list(params['shape'], ['height', 'width'])` for example.

Reply:
@mraspaud You are right that `params` isn't used more than that, but `var` is used a lot to know exactly what we are dealing with. Besides keeping it as is, another option would be to split the function up per-parameter but I think most of the functionality is shared so I'm not sure it would be pretty/easy.